### ISSUE TITLE
_generateTraceId() throws RangeError on Flutter Web

### ISSUE THREAD

**Issue by francescoberardi** [DataDog/dd-sdk-flutter#558]
**Describe what happened**

On Flutter web _generateTraceId() method throws `Uncaught Error: RangeError: max must be in range 0 < max ≤ 2^32, was 0`.
This happen because shift operators 1 << 32 returns 0 in web and and the call:

`_traceRandom.nextInt(1 << 32) `

throws:

`Uncaught Error: RangeError: max must be in range 0 < max ≤ 2^32, was 0`

**Steps to reproduce the issue:**
Can be reproduced running this code on [dartpad](https://dartpad.dev/):
```
import 'dart:math';

void main() {
  print(1 << 32);
  final Random traceRandom = Random();
  traceRandom.nextInt(1 << 32);
}
```
or using datadog_gql_link on Flutter web

**Additional context**
 - Dart/Flutter version: 3.16.9
 - Flutter web
 - Datadog SDK version:   datadog_flutter_plugin: ^2.2.0 and  datadog_gql_link: ^1.0.0

Similar issue was reported for uuid library:

[uuid issue](https://github.com/daegalus/dart-uuid/issues/103)

they fixed replacing:

`var k = _secureRandom.nextInt(1 << 32);`

with

`var k = _secureRandom.nextInt(pow(2, 32).toInt());`

(https://github.com/daegalus/dart-uuid/commit/56aa462444230e3da1c550838d7f3197f050857f)

**Comment by fuzzybinary** [comment]
Thanks for reporting @francescoberardi. I'll take a look into it.

I feel like this is almost a Dart bug as well, but obviously we can't rely on it being fixed, so I'll get in the workaround.

Note -- I'm worried that  `gql_link` will produce double resources on Flutter web. You may have to tell me if it works or not once the patch is in.

**Comment by francescoberardi** [comment]
Thanks @fuzzybinary 

> Note -- I'm worried that gql_link will produce double resources on Flutter web. You may have to tell me if it works or not once the patch is in.

Sure 👍🏻

**Comment by fuzzybinary** [comment]
This should now be available in `datadog_flutter_plugin` 2.3.0. An update to the gql_link package shouldn't be necessary. Let me know if you have any issues!
